internal/olm/operator/registry/configmap/deployment_test.go

```diff
@@ -0,0 +1,229 @@
+			for _, p := range paths {
+				volumeMounts = append(volumeMounts, corev1.VolumeMount{
+					Name:      "testVolName",
+					MountPath: p,
+				})
+			}
+			for i := range dep2.Spec.Template.Spec.Containers {
+				dep2.Spec.Template.Spec.Containers[i].VolumeMounts = append(dep2.Spec.Template.Spec.Containers[i].VolumeMounts, volumeMounts...)
+			}
+			f := withContainerVolumeMounts("testVolName", paths...)
+			f(dep)
+
+			Expect(dep.Spec.Template.Spec.Containers).Should(Equal(dep2.Spec.Template.Spec.Containers))
+		})
+	})
+
+	Describe("getDBContainerCmd", func() {
+		It("should apply f to dep's pod template spec and apply volumes", func() {
+			initCmd := fmt.Sprintf("/bin/initializer -o %s -m %s", registryDBName, containerManifestsDir)
+			srvCmd := fmt.Sprintf("/bin/registry-server -d %s -t %s", registryDBName, registryLogFile)
+
+			Expect(getDBContainerCmd(registryDBName, registryLogFile)).Should(Equal(fmt.Sprintf("%s && %s", initCmd, srvCmd)))
+		})
+	})
+
+	Describe("withRegistryGRPCContainer", func() {
+		It("should apply f to dep's pod template spec and append contaiers", func() {
+			container := corev1.Container{
+				Name:       getRegistryServerName("testPkg"),
+				Image:      registryBaseImage,
+				WorkingDir: "/tmp",
+				Command:    []string{"/bin/sh"},
+				Args: []string{
+					"-c",
+					getDBContainerCmd(registryDBName, registryLogFile),
+				},
+				Ports: []corev1.ContainerPort{
+					{Name: "registry-grpc", ContainerPort: registryGRPCPort},
+				},
+			}
+			dep := &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Replicas: nil,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							NodeName:   "testNode",
+							Containers: nil,
+						},
+					},
+				},
+			}
+			dep2 := &appsv1.Deployment{
+				Spec: appsv1.DeploymentSpec{
+					Replicas: nil,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							NodeName:   "testNode2",
+							Containers: nil,
+						},
+					},
+				},
+			}
+
+			dep2.Spec.Template.Spec.Containers = append(dep2.Spec.Template.Spec.Containers, container)
+			f := withRegistryGRPCContainer("testPkg")
+			f(dep)
+
+			Expect(dep.Spec.Template.Spec.Containers).Should(Equal(dep2.Spec.Template.Spec.Containers))
+		})
+	})
+
+	Describe("newRegistryDeployment", func() {
+		It("should return a dployment", func() {
+			var replicas int32 = 1
+			dep := &appsv1.Deployment{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: appsv1.SchemeGroupVersion.String(),
+					Kind:       "Deployment",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      getRegistryServerName("testPkg"),
+					Namespace: "testns",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: getRegistryDeploymentLabels("testPkg"),
+					},
+					Replicas: &replicas,
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: getRegistryDeploymentLabels("testPkg"),
+						},
+					},
+				},
+			}
```

Comment:
I would move `dep` to `BeforeEach` statement for this test so we can reuse it in 2 tests.